processor/batchprocessor/splitmetrics.go

```diff
@@ -108,41 +108,56 @@ func splitMetric(ms, dest pdata.Metric, size int) (int, bool) {
 		return metricDataPointCount(ms), true
 	}
 
-	msSize, i := metricDataPointCount(ms)-size, 0
+	msSize, i, j := metricDataPointCount(ms)-size, 0, 0
 	filterDataPoints := func() bool { i++; return i <= msSize }
+	filterDestPoints := func() bool { j++; return j > size }
 	switch ms.DataType() {
 	case pdata.MetricDataTypeIntGauge:
-		dest.IntGauge().DataPoints().Resize(size)
+		dest.IntGauge().DataPoints().RemoveIf(func(_ pdata.IntDataPoint) bool {
+			return filterDestPoints()
+		})
 		ms.IntGauge().DataPoints().RemoveIf(func(_ pdata.IntDataPoint) bool {
 			return filterDataPoints()
 		})
 	case pdata.MetricDataTypeDoubleGauge:
-		dest.DoubleGauge().DataPoints().Resize(size)
+		dest.DoubleGauge().DataPoints().RemoveIf(func(_ pdata.DoubleDataPoint) bool {
+			return filterDestPoints()
+		})
```

Comment:
This looks pretty awkward and harder to understand. `Resize` seemed to be not a bad idea?

Reply:
Yeah, and I think this was the simpler one to understand.  It would be sad to have to keep `Resize` around just for this, but I'm not sure there is a better option.

Reply:
I rewrote this to avoid copying the entire metric when its datapoints are to be split.  I _think_ this was the only place where using `Resize()` would still be attractive from a code reduction standpoint, but this implementation is probably still preferable.